src/organizations/components/OrgProfileTab/index.tsx

```diff
@@ -16,42 +16,46 @@ import CopyableLabeledData from 'src/organizations/components/OrgProfileTab/Copy
 import DeletePanel from 'src/organizations/components/OrgProfileTab/DeletePanel'
 
 // Utils
-import {getOrg} from 'src/organizations/selectors'
 import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 import {CLOUD} from 'src/shared/constants'
+import {shouldUseQuartzIdentity} from 'src/identity/utils/shouldUseQuartzIdentity'
 
-// Types
+// Selectors
 import {getMe} from 'src/me/selectors'
+import {getOrg} from 'src/organizations/selectors'
+import {selectQuartzIdentity} from 'src/identity/selectors'
+
+// Thunks
+import {getCurrentOrgDetailsThunk} from 'src/identity/actions/thunks'
 
+// Styles
 import 'src/organizations/components/OrgProfileTab/style.scss'
-import {
-  getBillingProviderThunk,
-  getCurrentOrgDetailsThunk,
-} from 'src/identity/actions/thunks'
-import {shouldUseQuartzIdentity} from 'src/identity/utils/shouldUseQuartzIdentity'
 
 const OrgProfileTab: FC = () => {
   const me = useSelector(getMe)
   const org = useSelector(getOrg)
-  const dispatch = useDispatch()
+  const identity = useSelector(selectQuartzIdentity)
 
-  const expectQuartzData = CLOUD && isFlagEnabled('uiUnificationFlag')
+  const dispatch = useDispatch()
 
   useEffect(() => {
     if (CLOUD && shouldUseQuartzIdentity()) {
-      if (!me.quartzMe.billingProvider) {
-        dispatch(getBillingProviderThunk())
-      }
-      if (!me.quartzMe.regionCode) {
+      if (
+        !me.quartzMe.regionCode ||
+        !me.quartzMe.regionName ||
+        !identity.org.provider
+      ) {
         dispatch(getCurrentOrgDetailsThunk())
       }
     }
   }, []) // eslint-disable-line react-hooks/exhaustive-deps
 
+  const expectQuartzData = CLOUD && isFlagEnabled('uiUnificationFlag')
+
   const hasSomeQuartzOrgData =
-    me.quartzMe?.billingProvider ||
-    me.quartzMe?.regionCode ||
-    me.quartzMe?.regionName
+    identity.org.provider || me.quartzMe?.regionCode || me.quartzMe?.regionName
```

Comment:
Super minor, but let's make the use of optional chaining here consistent for both references to `identity.org` (so that it behaves the same as the references to `me.quartzMe`, and we don't have an uncaught error if `identity` does not have the org property for some reason).

`identity.org?.provider`